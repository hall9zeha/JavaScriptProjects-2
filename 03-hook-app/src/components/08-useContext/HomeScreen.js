import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
    //Buscará la instancia de UserContext en el árbol de componentes
    //ya que necesitamos acceder lo que hayamos compartido a través de ella
    const {user} = useContext(UserContext)
    console.log(user);
  return (
    <div>
        <h1>HomeScreen</h1>
        <pre>
          {JSON.stringify(user,null,3)}
        </pre>
        <hr/>
    </div>
  )
}
