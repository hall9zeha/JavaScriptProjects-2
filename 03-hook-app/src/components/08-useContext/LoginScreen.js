import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    const {user, setUser} = useContext(UserContext);

  
        const {id,name,email} = user;
        setUser({
            id,
            name,
            email,
            sigIn:true
        })
  

  return (
    <div>
        <h1>LoginScreen</h1>
        <hr/>
        <button onClick={setUser} className='btn btn-primary '>Login</button>
    </div>
  )
}
