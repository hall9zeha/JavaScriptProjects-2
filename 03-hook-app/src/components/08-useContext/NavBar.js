import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm  navbar-dark  bg-dark">
            <div className="container-fluid">
                <a className="navbar-brand" href="#">useContext</a>
              
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    {/* NavLink nos permite usar algunas clases de bootstrap */}
                    <NavLink exact activeClassName="active" className="nav-link nav-item " to='/'>Home</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link nav-item" to='/about'>About</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link nav-item" to='/login'>Login</NavLink>
                 
                </div>
                </div>
            </div>
        </nav>
   
  )
}
