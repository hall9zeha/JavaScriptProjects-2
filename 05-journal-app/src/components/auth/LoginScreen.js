import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import { useDispatch } from 'react-redux'
import { login, startLoginAsyncExample, startLoginWithGoogle } from '../../actions/auth'

export const LoginScreen = () => {

  const dispatch = useDispatch()


  const [formValues, handleInputChange ] = useForm({
    email:'martha@gmail.com',
    password:12345678
  })

  const {email,password} = formValues

  const handleEventLogin = (e) =>{
    e.preventDefault();
    dispatch(login(123456,'Martha'))
    
    // Ejemplo de acción asíncrona para probar que la configuración middleware en redux funciona correctamente
    // dispatch(startLoginAsyncExample(email,password));
  }
  const handleLoginWithGoogle =()=>{
    dispatch(startLoginWithGoogle());
  }
  return (
    <>
      <h3 className='auth__title'>Login</h3>
      <form onSubmit={handleEventLogin}>
        <input
          type='text'
          placeholder='Email'
          name="email"
          autoComplete='off'
          className='auth__input'
          value={email}
          onChange={handleInputChange}
        />
        <input
          type='password'
          placeholder='Password'
          name='password'
          className='auth__input'
          value={password}
          onChange={handleInputChange}
        />
        <button

          type='submit'
          className='btn btn-primary btn-block'
        >
          Login
        </button>
    
        <div className='auth__social-networks'>
          <p>Login with social networks</p>
              <div  className="google-btn" 
                    onClick={handleLoginWithGoogle}
              >
                  <div className="google-icon-wrapper">
                      <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/archive/c/c1/20210618182605%21Google_%22G%22_logo.svg" alt="google button" />
                  </div>
                  <p className="btn-text">
                      <b>Sign in with google</b>
                  </p>
              </div>
        </div>
        <Link
          to="/auth/register"
          className="link"
        >
          Create new account
        </Link>   
      </form>
    </>
  )
}
