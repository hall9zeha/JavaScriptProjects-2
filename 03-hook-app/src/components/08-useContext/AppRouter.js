import React from 'react'
import { BrowserRouter as  Router, Routes, Route } from 'react-router-dom'
import { AboutScreen } from './AboutScreen'
import { HomeScreen } from './HomeScreen'
import { LoginScreen } from './LoginScreen'
import { NavBar } from './NavBar'

export const AppRouter = () => {
  return (
    <Router>
        <div>
            <NavBar/>
            {/* El Routes de react-router es como un switch tradicional(Switch en react-router-dom está descontinuado), dependiendo de qué condición ejecutará tal o cual
            En este caso una dirección a un componente dentro de la ruta */}
            <Routes>
            {/* exact le dice a la ruta que debe buscar el path con todos sus caracteres y no por la primera coincidencia */}
                <Route exact path='/' Component={HomeScreen}/> 
                <Route exact path='/login' Component={LoginScreen}/>
                <Route exact path='/about' Component={AboutScreen}/>
                {/* Ponemos una ruta por defecto si no existiera ninguna de las anteriores */}
                <Route path='*' Component={HomeScreen}/> 
            </Routes>
        </div>
    </Router>
  )
}
