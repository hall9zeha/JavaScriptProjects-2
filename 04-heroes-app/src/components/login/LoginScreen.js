import React, { useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import { AuthContext } from '../../auth/authContext';
import { Types } from '../../types/Types';

export const LoginScreen = () => {
    const navigate = useNavigate();

    const {dispatch} = useContext(AuthContext);

    const handleLogin=()=>{
      const action = {
        type:Types.login,
        payload:{name:'Martha'}
      }
      dispatch(action);
      navigate('/marvel',{
          replace:true
      })
    }

  return (
    <div className='container mt-5'>
        <h1>Login</h1>
        <hr/>

        <button className='btn btn-primary ' onClick={handleLogin}>Login</button>
    </div>
  )
}
