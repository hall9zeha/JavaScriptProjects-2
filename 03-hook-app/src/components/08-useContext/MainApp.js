import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
    const userProvides = {
        id:123,
        name:"Martha",
        email:"martha@mail.com",
        sigIn:false
    }

    const [userState, setUserState] = useState(userProvides);

  return (
    //Envolver nuestros componentes con UseProvider nos permitir√° compartir 
    //elementos u objetos con sus componentes hijos 

    //Ahora enviaremos un objeto que contenga las propiedades de nuestro useState
    <UserContext.Provider value={
        {user:userState,
         setUser:setUserState
        }}>
        <AppRouter/>
    </UserContext.Provider>
  )
}

