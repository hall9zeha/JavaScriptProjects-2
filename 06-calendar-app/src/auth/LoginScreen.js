import React from 'react';
import './login.css';
import { useForm } from '../hooks/useForm';
import { useDispatch } from 'react-redux';
import { startLogin, startRegister } from '../actions/auth';
import Swal from 'sweetalert2';

export const LoginScreen = () => {
    const dispatch = useDispatch();

    //For login
    const [formLoginValues, handleLoginInputchange] = useForm({
        lEmail:'barry@mail.com',
        lPassword:'123456'
    })

    const {lEmail, lPassword} = formLoginValues;

    //For register

    const [formRegisterValues, handleRegisterInputChange] = useForm({
        rName:'Lisa',
        rEmail:'lisa@mail.com',
        rPassword:'123456',
        rPassword2:'123456'
    })
    const {rName,rEmail,rPassword,rPassword2} = formRegisterValues;

    const handleLogin =(e)=>{
        e.preventDefault()
       
        dispatch(startLogin(lEmail,lPassword))
    }

    const handleRegister = (e)=>{
        e.preventDefault();
        if(rPassword !== rPassword2){
            Swal.fire('Error','Passwords must be the same');
        }
        dispatch(startRegister(rName,rEmail,rPassword))
    }

    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-1">
                    <h3>Ingreso</h3>
                    <form onSubmit={handleLogin}>
                        <div className="form-group mb-2">
                            <input 
                                type="text"
                                className="form-control"
                                placeholder="Correo"
                                name='lEmail'
                                value={lEmail}
                                onChange={handleLoginInputchange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contraseña"
                                name='lPassword'
                                value={lPassword}
                                onChange={handleLoginInputchange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input 
                                type="submit"
                                className="btnSubmit"
                                value="Login" 
                            />
                        </div>
                    </form>
                </div>

                <div className="col-md-6 login-form-2">
                    <h3>Registro</h3>
                    <form onSubmit={handleRegister}>
                        <div className="form-group mb-2">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Nombre"
                                name='rName'
                                value={rName}
                                onChange={handleRegisterInputChange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name='rEmail'
                                value={rEmail}
                                onChange={handleRegisterInputChange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contraseña" 
                                name='rPassword'
                                value={rPassword}
                                onChange={handleRegisterInputChange}
                            />
                        </div>

                        <div className="form-group mb-2">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la contraseña"
                                name='rPassword2'
                                value={rPassword2}
                                onChange={handleRegisterInputChange} 
                            />
                        </div>

                        <div className="form-group mb-2">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Crear cuenta" />
                        </div>
                    </form>
                </div>
            </div>
        </div>
    )
}