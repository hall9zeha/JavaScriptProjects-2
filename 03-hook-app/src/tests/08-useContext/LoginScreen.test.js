import React from 'react';

import renderer from 'react-test-renderer';
import { LoginScreen } from '../../components/08-useContext/LoginScreen';
import { UserContext } from '../../components/08-useContext/UserContext';

describe('Tests in <LoginScreen/> component', () => { 

    const setUserMock = jest.fn();
    

    const userMock={
        id:2,
        name:'Martha',
        email:'martha@mail.com',
        sigIn:true
    }
    const wrapper = renderer.create(
        <UserContext.Provider value={{user:userMock, setUser:setUserMock}}>
            <LoginScreen/>
        </UserContext.Provider>);
        

    test('should show successfully', () => { 
        expect(wrapper).toMatchSnapshot();

     })

    test('should execute setUser function with same expected arg', async()=>{
      
        const container = renderer.create(
            <UserContext.Provider value={{user:userMock, setUser:setUserMock}}>
                <LoginScreen/>
            </UserContext.Provider>);
        await  container.root.findByType('button').props.onClick();
        expect(setUserMock).toHaveBeenCalledTimes(2)
        expect(setUserMock).toHaveBeenCalledWith(userMock)
      
    })
 })