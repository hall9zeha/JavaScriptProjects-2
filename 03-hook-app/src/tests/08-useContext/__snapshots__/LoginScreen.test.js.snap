// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <LoginScreen/> component should show successfully 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary "
    onClick={[MockFunction]}
  >
    Login
  </button>
</div>
`;
